pkg/i2gw/ingress2gateway_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package i2gw
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	networkingv1 "k8s.io/api/networking/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_inputFile(t *testing.T) {
+	iPrefix := networkingv1.PathTypePrefix
+	ingress1ClassName := "ingress1-example"
+	ingress2ClassName := "ingress2-example"
+
+	expectIngresses := []networkingv1.Ingress{
```

Comment:
I am not against having this list but I think it would be better to have it as a map where the name is describing the ingress you expect like: `ingress_with_test1_backend_exposed_in_443` (you could probably think of a better name that describes what you are expecting) and the value is the ingress.

Reply:
Alternatively, instead of looping over them when you compare, have a function that checks whether the `expectedIngresses` are the same as what you got in the `ingressList`

Reply:
I made a helper method that does the internal loop